src/Toml/Printer.hs

```diff
@@ -171,13 +173,17 @@ prettyTableArrays options i pref = mapOrdered arrText options
 
 -- Returns an indentation prefix
 tabWith :: PrintOptions -> Int -> Text
-tabWith options n = Text.replicate (n * indent options) " "
+tabWith options n = Text.replicate (n * printOptionsIndent options) " "
 
 -- Returns a proper sorting function
-mapOrdered :: Ord k => ((k, v) -> [t]) -> PrintOptions -> HashMap k v -> [t]
-mapOrdered f options
-    | shouldSort options = concatMap f . sortOn fst . HashMap.toList
-    | otherwise          = concatMap f . HashMap.toList
+mapOrdered :: ((Key, v) -> [t]) -> PrintOptions -> HashMap Key v -> [t]
```

Comment:
Okay, I see the problem now. I don't think we should introduce a separate comparator for `Key` and `Piece`, because `Piece` can be considered as a special case of `Key`. So here in this function we can pass a list of type `[(Key, v)]` instead of `HashMap Key v` and convert to this list on all call-sites.

Reply:
Awesome - this was exactly the insight I was needing! Let me work on fleshing it out :)